include/RAJA/exec-agency/forall_agency.hxx

```diff
@@ -136,6 +136,51 @@ namespace RAJA
                         });
   }
 
+template <typename SEG_EXEC_POLICY, typename LOOP_BODY>
+RAJA_INLINE void forall(
+    IndexSet::ExecPolicy<agency_taskgraph_segit, SEG_EXEC_POLICY>,
+    const IndexSet& iset,
+    LOOP_BODY loop_body)
+{
+  if (!iset.dependencyGraphSet()) {
+    std::cerr << "\n RAJA IndexSet dependency graph not set "
+              << "FILE: " << __FILE__ << " line: " << __LINE__ << std::endl;
+    exit(1);
+  }
+
+  // Cast away const because we want it in the other overloaded versions,
+  // but need to modify the index sets due to how dependent nodes are
+  // implemented.
+  IndexSet* ncis = const_cast<IndexSet*>(&iset);
+  int numSegments = ncis->getNumSegments();
+  
+  agency::bulk_invoke(agency::par(numSegments), [=] (agency::parallel_agent& self) {
+    int isi = self.index();
```

Comment:
It looks like a ton of the code in here is almost identical to the OMP version; should we push that out into a separate helper function that is templated on execution policy, so if we wanted to we could specialize but as long as they're the same just leave it alone?